docs/knowledgebase/advanced/consensus.md

```diff
@@ -20,7 +23,7 @@ Blockchains must agree on:
 - A final (current) state.
 
 In order to agree on the resulting state after a transition, all operations within a blockchain's
-[state transition function](../runtime/index) must be deterministic.
+[state transition function](../runtime/index) must be deterministic. In order to be deterministic, the same operation performed across different nodes should return the same result. A difference in results between the nodes for the same operation can output a failure in consensus as storing data on the ledger will lead to an inconsistent ledger state, thereby making the whole smart contract unusable.
```

Comment:
> can output a failure in consensus as storing data on the ledger will lead to an inconsistent ledger state, thereby making the whole smart contract unusable

I don't really understand the phrase "can output a failure in consensus" and I don't see why smart contracts are relevant here

Reply:
Indeed, a consensus failure is not equivalent to conflicting opinions on state. In fact, consensus is made to resolve conflicting opinions on state.